src/axom/sidre/core/Group.cpp

```diff
@@ -2292,42 +2293,42 @@ Group* Group::walkPath(std::string& path, bool create_groups_in_path)
 {
   Group* group_ptr = this;
 
-  std::string::size_type pos = detail::find_exclusive(path, s_path_delimiter);
-  if(pos != std::string::npos)
-  {
-    std::vector<std::string> tokens = detail::split(path, s_path_delimiter, pos);
-    std::vector<std::string>::iterator stop = tokens.end() - 1;
+  const std::vector<std::string> path_parts = axom::utilities::Path(path, s_path_delimiter).parts();
 
-    // Navigate path down to desired Group
-    for(std::vector<std::string>::const_iterator iter = tokens.begin();
-        iter < stop;
-        ++iter)
+  // Find stopping point (right before last part of path)
+  std::vector<std::string>::const_iterator stop = path_parts.end() - 1;
+
+  // Navigate path down to desired Group
+  for(std::vector<std::string>::const_iterator iter = path_parts.begin();
+      iter < stop;
+      ++iter)
+  {
+    if(*iter == "")
```

Comment:
The previous split was ignoring empty tokens, so that's why I remove the previous assert for just skipping them when the new split returns them.